force-app/main/default/classes/ContactController.cls

```diff
@@ -1,170 +1,132 @@
+            List<Contact> source = (List<Contact>) JSON.deserialize(frontSource, List<Contact>.class);
+            List<Case> updateCases = new List<Case>();
+            List<Id> accountsId = getAccountsId(source);
+            Map <Id, List<Contact>> accountsWithContacts = getColleagues(source, accountsId);
+            List<Id> allContactsId = getAllContactsId(source, accountsWithContacts);
+            Map<Contact, List<Case>> contactsWithCases = getAllCases(allContactsId);            
+            Map<Contact, List<Case>> delContactsWithCases = getSpecificContactsWithCases(true, contactsWithCases, source);
+            Map<Contact,List<Case>> stayContactsWithCases = getSpecificContactsWithCases(false, contactsWithCases, source);
+            for(Contact c : delContactsWithCases.keySet()){
+                if(delContactsWithCases.get(c).size() == 0){
+                    continue;
+                }else {
+                    List<Contact> colleagues = accountsWithContacts.get(c.AccountId);
+                    if (colleagues == null) {
+                        AuraHandledException e = new AuraHandledException('');
+                        e.setMessage('first error: DELETE_FAILED, Your attempt to delete this contact could not be completed because it is associated with some cases');
+                        throw e;
+                    } else {
+                        Contact newExecutor = getNewExecutor(colleagues, stayContactsWithCases);
+                        for(Case cs : delContactsWithCases.get(c)){
+                            cs.ContactId = newExecutor.Id;
+                            updateCases.add(cs);
                         }
-                        delCases.addAll(closedCases);
-                        System.debug('delCases: ' + delCases);
-                        delContact = new Contact(Id = contactId);                 
-                        delContacts.add(delContact); 
                     }
-                }  
-                if(!delCases.isEmpty()){
-                    delete new List<Case>(delCases);
-                }
-                if (!updatedCases.isEmpty()) {
-                    update new List<Case>(updatedCases);
-                } 
-                delete delContact;                      
-            }                       
-            return delContacts;
-        } catch (Exception e) {
-            throw new AuraHandledException(' delete Contacts' + e.getMessage() + e.getStackTraceString());
-        }    
-    }
-    
-    @AuraEnabled
-    public static List<Contact> findContactColleagues(Id contactId){
-        try {
-            Account contactAccount = findAccountByContactId(contactId);
-            return [
-                SELECT Id FROM Contact WHERE AccountId = :contactAccount.Id AND Id != :contactId
-            ];          
+                }                
+            } 
+            update updateCases;
+            delete source; 
+            maxCases = 0;    
```

Comment:
I am not sure if this is needed unless you are going to use this method somewhere else, I mean every new calling this method from UI it is completely new transaction and maxCases variable should be initialized in 3rd line again (with 0 value)

Reply:
Fixed